### PR TITLE
Introduce expression converters

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: src/QueryExpression/Converter/Chain.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace DigitalBackstage\Searchable\QueryExpression\Converter;
+
+use DigitalBackstage\Searchable\QueryExpression;
+use DigitalBackstage\Searchable\QueryExpression\Converter;
```
Comment (achevillard) [inline src/QueryExpression/Converter/Chain.php]: useless => QueryExpression\Converter

File: src/QueryExpression/Converter.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+use DigitalBackstage\Searchable\QueryExpression;
+
+/**
+ * Converts a query expression into a storage-specific equivalent
+ */
+interface Converter
+{
+    public function convert(QueryExpression $queryExpression);
+
+    public function supports(QueryExpression $queryExpression): bool;
```
Comment (achevillard) [inline src/QueryExpression/Converter.php]: throws \RuntimeException

File: src/QueryExpression/Converter/Functional.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace DigitalBackstage\Searchable\QueryExpression\Converter;
+
+use DigitalBackstage\Searchable\QueryExpression;
+use DigitalBackstage\Searchable\QueryExpression\Converter;
```
Comment (achevillard) [inline src/QueryExpression/Converter/Functional.php]: useless

File: src/QueryExpression/Converter/Functional.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace DigitalBackstage\Searchable\QueryExpression\Converter;
+
+use DigitalBackstage\Searchable\QueryExpression;
+
+final class Functional implements QueryExpression\Converter
```
Comment (Arnch) [inline src/QueryExpression/Converter/Functional.php]: Converter

**Review by achevillard — APPROVED** [review]

**Pull request merged**